daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -315,22 +331,18 @@ object Speedy {
       }
     }
 
-    def enterFullyAppliedFunction(prim: Prim, args: util.ArrayList[SValue]): Unit = {
+    def enterFullyAppliedFunction(
```

Comment:
Yes. There is potentially some refactoring which will improve things here. But I am reluctant to make the obvious change... i.e. only passing `this` for the `KFun`/`Frame`, because the callee will then have to pull the components out. Maybe the `frame` should be set by the caller of `enterFullyAppliedFunction` - that might be cleaner.

Reply:
I'm afraid that if we don't clean this duplication up in this PR it might never happen. It shouldn't be too much of a change, is it?

Reply:
I changed so frame is set by the caller.